core/datastorage/src/main/java/com/cognifide/aet/vs/metadata/MetadataDAOMongoDBImpl.java

```diff
@@ -135,12 +152,26 @@ public Suite getLatestRun(DBKey dbKey, String name) throws StorageException {
         .find()
         .sort(Sorts.descending(SUITE_VERSION_PARAM_NAME));
 
-    return FluentIterable.from(found).transform(new Function<Document, Suite>() {
-      @Override
-      public Suite apply(Document result) {
-        return new DocumentConverter(result).toSuite();
-      }
-    }).toList();
+    return StreamSupport.stream(found.spliterator(), false)
+        .map((Function<Document, Suite>) document -> new DocumentConverter(document).toSuite())
+        .collect(Collectors.toList());
+  }
+
+  public List<List<String>> listSuiteVersions(DBKey dbKey, String name) throws StorageException {
```

Comment:
This method should return `List<Map<String, String>>` or `List<SuiteVersion>` (where `SuiteVersion` is a new class with 2 fields - `version` and `correlationId`. Then in `HistoryServlet` you just need to call `gson.toJson` instead of the whole `JsonArray`/`JsonObject` building